packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -44,6 +44,9 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
     assert(renderView != null);
     addPersistentFrameCallback(_handlePersistentFrameCallback);
     initMouseTracker();
+    if (kIsWeb) {
```

Comment:
How does this behave with hot restart? Does it get called again?

Reply:
it does get called again, but now (thanks to @yjbanov ) in run mode we definitely do not create a real service worker so it would only reload an empty script.